### ISSUE TITLE
VariableHeader sp data exceeds header buffer size!   various controllers acting like channels crossed at multiple times

### ISSUE THREAD

**Issue by roofrat** [FalconChristmas/fpp#886]
**NOTE:** IF YOU DO NOT ATTACH YOUR LOG FILES, THEN THERE IS A HIGHER PROBABILITY THE ISSUE WON'T BE RESOLVED.  (Steps at end) 
# Please answer the following:
**FPP 4.51 running on pi 3B+*

**Describe the bug and Steps to reproduce**: 
A clear and concise description of what the bug is and the steps to be taken to reproduce. 
1. start my show sequence
2. checked logs because of errors I was having before the update  which I'm still having.  I have 2 f16's and 1 f48.  one f16 works fine, the other f16 and f48 randomly decide one of them is going to make a few props look like date channels are crossed.  Only 1 controller at a time does this not both at the same time.  This causes props on different ports to either go out completely or have locked up pixels that stay on or won't properly display as commanded to
running pixels on test mode they all work perfectly
running sequences individually they run perfectly
running a few sequences the propblems happen

3. observed this errr

2020-11-28 14:21:50.663 (3051) fppd.c:752: Starting main processing loop
2020-11-28 14:22:05.383 (3051) fseq/FSEQFile.cpp:493: VariableHeader sp data exceeds header buffer size!2020-11-28 14:25:25.052 (3051) fseq/FSEQFile.cpp:493: VariableHeader sp data exceeds header buffer size!
**Expected behavior**: 
not sure what is giving the buffer error.  

**Additional context**: 
multiple days in zoom room, can't figure out why certain props on different controllers will degrade.  
[FPP_Logs_20201128.1927.zip](https://github.com/FalconChristmas/fpp/files/5611138/FPP_Logs_20201128.1927.zip)
[FPP_Logs_20201128.1927.zip](https://github.com/FalconChristmas/fpp/files/5611139/FPP_Logs_20201128.1927.zip)
[FPP_Logs_20201128.1927.zip](https://github.com/FalconChristmas/fpp/files/5611140/FPP_Logs_20201128.1927.zip)
[FPP_Logs_20201128.1927.zip](https://github.com/FalconChristmas/fpp/files/5611142/FPP_Logs_20201128.1927.zip)

**Additional Attachments**
Please include the log files. To create the file:
1. Open the FPP UI
2. Open the the File Manager (Content Setup->File Manager), 
3. Select the "Logs" tab
4. Click "zip" button.
5. Attach to this ticket.

**Comment by jcrossbdn** [comment]
Interesting I also see this in my fppd logs but after fixing a channel overlap I no longer have jittering in my show when it happens.

**Comment by patdelaney** [comment]
Jason please note that "Excessive" logging can cause lag if FPP is
writing a lot of data back to the disk. If everything is now working as
expected please return logging to normal.   Just a heads up

**Comment by jcrossbdn** [comment]
@patdelaney in my specific case logging is set to normal but I didn't see the error again after a few reboots during last nights dry run so it could have been a few different issues manifesting that warning.

**Comment by Mat-Moo** [comment]
I've got the same error in my logs:-
2020-12-02 16:18:39.502 (23448) fseq/FSEQFile.cpp:493: VariableHeader sp data exceeds header buffer size!
2020-12-02 16:21:01.815 (23448) fseq/FSEQFile.cpp:493: VariableHeader sp data exceeds header buffer size!  
![image](https://user-images.githubusercontent.com/908888/100900365-b0d14c00-34ba-11eb-9c97-f2f8a74f5562.png)
Don't think I have any overlaps.
Is this message that the source data is too long? eg the sequence software setup wrong (in my case Vixen)

**Comment by czajkaej** [comment]
I have a similar error coming up, when running on 4.5 code. When I run my show, it plays part way thru a song and it seems that the FPPd crashes partially. It stops sending e1.31 packets out to my pixel controller.

I have my Master FPP running 3.6 stable with same configuration, but when I tried 4.5 again, I got the crashing data stream. All of my remote FPP continue playing, but the e1.31 stops. I moved my e1.31 to a remote, but my master still crashes, and won't start playing songs until I restart service manually via UI.

Dec  6 17:23:21 FPP-Main fppd[1292]: 2020-12-06 17:21:42.544 (1292) fseq/FSEQFile.cpp:493: VariableHeader sp data exceeds header buffer size!2020-12-06 17:23:12.842 (1476) MultiSync.cpp:1619: Error: Unable to send multisync packet: Resource temporarily unavailable   (-1/5)
Dec  6 17:23:21 FPP-Main fppd[1292]: 2020-12-06 17:23:13.049 (1292) MultiSync.cpp:1619: Error: Unable to send multisync packet: Resource temporarily unavailable   (-1/5)

[fppd.log](https://github.com/FalconChristmas/fpp/files/5650609/fppd.log)

**Comment by nmiller0113** [comment]
@patdelaney @dkulp @cpinkham - I'm seeing this same VariableHeader message in my logs but it appears only for a single sequence, and it happens every single time I play it, though the sequence plays as it should without any visible issues or crashes. This sequence is transfered to my master and remote F32B Kulp Controllers via FPP Connect in xLights, and all the files are sparse fseq's, none are full. The main FSEQ is quite large ~30mb, so I'm [attaching a link to download the zip](https://drive.google.com/file/d/13YzWv5wFdvj4z9OiT1qqHMp2ThKon1oI/view?usp=sharing) of it and the xsq file. You should be able to use that to reproduce it every time. In case it is important, this sequence was made with xLights 2020.51 for Mac. Thanks!

**Comment by czajkaej** [comment]
Hmmm, I am also sequencing on Mac, but I am uploading my files in V2 format.

**Comment by nmiller0113** [comment]
@czajkaej so we have two parts in common. We're both running V2 fseq's, but mine are sparse to all of my controllers, and we're both building it from the Mac version of xLights. This error seems like a bad fseq since it doesn't happen with all of them, which makes me lean towards the possibility of this potentially being an xLights issue. I'll keep the report to just fpp for now since @dkulp is also an xLights developer and would be able to make a simple crossover if needed.

**Comment by patdelaney** [comment]
Can either of you create that same sequence that is giving you trouble on a
windows system and test it?

**Comment by libby2cm** [comment]
I bunch of my remotes had the same error, all are using sparse v2 fseq's.. however, since I patched to 4.5.1.6 the other night, none of them seem to be doing it anymore.

**Comment by nmiller0113** [comment]
It looks like @dkulp corrected it [here](https://github.com/FalconChristmas/fpp/commit/04b1df87e173424637034c09e1dec6a6e6ed6ee0)

I'm no expert but it looks like it was a logging correction and not an actual FSEQ file issue. @patdelaney more than happy to create the same sequence in Windows but since I updated my FPP instances that error went away, so I'm not sure it's necessary.

**Comment by patdelaney** [comment]
no reason to now, but thanks

**Comment by czajkaej** [comment]
I’ll run a test tomorrow with most current version

**Comment by czajkaej** [comment]
I switched over to the latest build, and noticed a few things. I was seeing errors in my log indicating voltage was back to normal. Keep in mind that 3.6 ran show perfectly fine, but I swapped out my power pack for 4.5.1 and it seemed to get more stable (less wifi drops, and FPPd stabilized too). To get past my dropping WiFi, I hung an extra line off my firewall and allowed show network to use the wired Ethernet to reach the internet for time and Remote Falcon. 

I am still running V2 sequences from my Mac, which never seemed to be the origin of any errors. It may have been the wall wart USB pack that the newer kernel didn’t like.
